extension/static/scripts/debugpy/memviz_get_variables_info.py

```diff
@@ -2,20 +2,53 @@
 import dataclasses
 import json
 import inspect
+import itertools
 from types import FunctionType, ModuleType
 from typing import Any, Callable, Dict, List, Optional, Tuple
 from sys import getsizeof
 from gc import get_referents
+from weakref import WeakValueDictionary
 
 
 PythonId = str
 FrameIndex = int
-ValueAccessExpr = str
 
 
 RETURN_VALUES_DICT_NAME = "__pydevd_ret_val_dict"
 
 
+class IdMap:
+    _weakrefMap: WeakValueDictionary[PythonId, Any] = WeakValueDictionary()
+    _strongrefMap: Dict[PythonId, Any] = {}
+
+    _NOT_FOUND = object()
+
+    @classmethod
+    def register(cls, python_id: PythonId, value: Any) -> None:
```

Comment:
I wonder if this is thread-safe :thinking: But I guess we shouldn't ever have more than one thread stopped and being debugged.